packages/calcite-components/src/components/meter/meter.tsx

```diff
@@ -0,0 +1,291 @@
+    const { low, high, min, max, value } = this;
+    const lowPercent = (100 * (low - min)) / (max - min);
+    const highPercent = (100 * (high - min)) / (max - min);
+    const currentPercent = (100 * (value - min)) / (max - min);
+    this.lowPercent = Math.round(lowPercent);
+    this.highPercent = Math.round(highPercent);
+    this.currentPercent = value ? Math.round(currentPercent) : 0;
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   *
+   * @param position
+   * @param value
+   * @param line
+   * @param label
+   * @param isValue
+   * @returns
+   */
+
+  private renderRange(
+    position: number,
+    value: string,
+    line?: boolean,
+    label?: string,
+    isValue?: boolean
+  ): VNode {
+    const labelClass = isValue ? CSS.meterLabelValue : CSS.meterLabelRange;
+    const style = { insetInlineStart: `${position}%` };
+    return (
+      <Fragment>
+        {line && (
+          <div
+            class={{ [CSS.meterStepLine]: true, [CSS.meterLabelContainer]: true }}
+            style={style}
+          />
+        )}
+        {(this.displayRangeLabels || isValue) && (
+          <div class={{ [CSS.meterLabel]: true, [labelClass]: true }} style={style}>
+            {value}
+            {label && this.unitLabel && this.labelType !== "percent" && (
+              <span class={CSS.meterUnitLabel}>&nbsp;{label}</span>
+            )}
+          </div>
+        )}
+      </Fragment>
+    );
+  }
+
+  render(): VNode {
+    const {
+      appearanceType,
+      currentPercent,
+      displayRangeLabels,
+      displayValue,
+      fillType,
+      high,
+      highPercent,
+      label,
+      labelType,
+      low,
+      lowPercent,
+      max,
+      maxPercent,
+      min,
+      minPercent,
+      unitLabel,
+      value,
+    } = this;
+    const kind = this.getMeterKind();
+    const isPercent = labelType === "percent";
+    const labelMin = this.formatLabel(isPercent ? minPercent : min);
+    const labelMax = this.formatLabel(isPercent ? maxPercent / 100 : max);
+    const labelLow = low ? this.formatLabel(isPercent ? lowPercent / 100 : low) : undefined;
+    const labelHigh = high ? this.formatLabel(isPercent ? highPercent / 100 : high) : undefined;
+    const labelValue = this.formatLabel(isPercent ? currentPercent / 100 : value || 0);
+    const valuePosition = currentPercent > 100 ? 100 : currentPercent > 0 ? currentPercent : 0;
+    const displayLow = !!low && low > value && low > min && (!high || low < high);
+    const displayHigh = !!high && high > value && high < max && (!low || high > low);
+
+    return (
+      <Host>
+        <div
+          aria-label={label}
+          aria-valuemax={isPercent ? maxPercent : max}
+          aria-valuemin={isPercent ? minPercent : min}
+          aria-valuenow={isPercent ? currentPercent : value}
```

Comment:
For some reason `aria-valuenow`/`aria-valuetext` isn't getting read back to AT, will need to dive into why this could be occurring tomorrow.

Reply:
Was able to diagnose the issue, the `aria-valuetext` is interfering with the `aria-valuenow` context for AT. If needed for context, could we add the `aria-valuetext`  value as part of `aria-label` so the `aria-valuenow` is provided to AT as the value changes?

Reply:
I think we can add the value to the `valuenow` to have it read as "50 percent", or "2,400 credits" (from slider example: https://www.w3.org/WAI/ARIA/apg/patterns/slider/examples/slider-temperature/)

Reply:
For what we're aiming for this looks good to go a11y-wise. 🚀 

NVDA does lose a "beep" indicator that content has changed in the page (when only `aria-valuenow` is provided over `aria-valuetext`). But that seems minor so the unit context is provided to AT. Nice work! 💪🏻

Reply:
Does the above case occur when `labelType="unit"` and there is no `unitLabel` provided? 

If so, we could consider having a translated default of "units" if we always want `aria-valuetext` to have some value?